fem/transfer.hpp

```diff
@@ -220,18 +283,34 @@ class L2ProjectionGridTransfer : public GridTransfer
       // arrays. The entries of the i'th high-order element are stored at the
       // index given by offsets[i].
       mutable Array<real_t> R, P;
-      Array<int> offsets;
+      mutable Array<real_t> R_ea, P_ea;
+
+      const bool use_device, verify_solution;
 
    public:
       L2ProjectionL2Space(const FiniteElementSpace& fes_ho_,
-                          const FiniteElementSpace& fes_lor_);
+                          const FiniteElementSpace& fes_lor_,
+                          const bool use_device_,
+                          const bool verify_solution_,
+                          Coefficient* coeff_,
+                          MemoryType d_mt_ = MemoryType::HOST);
+
+      /*Same as above but assembles and stores R_ea, P_ea */
+      void DeviceL2ProjectionL2Space(const FiniteElementSpace& fes_ho_,
+                                     const FiniteElementSpace& fes_lor_,
+                                     const Coefficient* coeff_);
```

Comment:
The parameters of this method are unnecessary -- they are already available as class members.